pkg/controllers/nodepool/readiness/controller.go

```diff
@@ -54,20 +54,29 @@ func NewController(kubeClient client.Client, cloudProvider cloudprovider.CloudPr
 func (c *Controller) Reconcile(ctx context.Context, nodePool *v1.NodePool) (reconcile.Result, error) {
 	ctx = injection.WithControllerName(ctx, "nodepool.readiness")
 	stored := nodePool.DeepCopy()
-	supportedNC := c.cloudProvider.GetSupportedNodeClasses()
 
-	if len(supportedNC) == 0 {
-		logger.Fatal("no supported node classes found for the cloud provider")
+	nodeClass, ok := lo.Find(c.cloudProvider.GetSupportedNodeClasses(), func(nc status.Object) bool {
+		return object.GVK(nc).GroupKind() == nodePool.Spec.Template.Spec.NodeClassRef.GroupKind()
+	})
+	if !ok {
+		// Ignore NodePools which aren't using a supported NodeClass.
+		// Note: should be unreachable due to predicate in controller reconciliation
+		return reconcile.Result{}, nil
 	}
-	nodeClass, err := c.getNodeClass(ctx, nodePool, supportedNC)
-	if err != nil {
+	err := c.kubeClient.Get(ctx, client.ObjectKey{Name: nodePool.Spec.Template.Spec.NodeClassRef.Name}, nodeClass)
+	if err != nil && !errors.IsNotFound(err) {
```

Comment:
```suggestion
	if err != nil {
```

Reply:
We do want to ignore not found errors here, if the NodeClass is not found we want to mark the NodePool as unready.